certbot/compat/filesystem.py

```diff
@@ -0,0 +1,125 @@
+"""Compat module to handle files security on Windows and Linux"""
+from __future__ import absolute_import
+
+import os  # pylint: disable=os-module-forbidden
+import stat
+
+try:
+    import ntsecuritycon  # pylint: disable=import-error
+except ImportError:  # pragma: no cover
+    ntsecuritycon = None  # type: ignore
+try:
+    import win32security  # pylint: disable=import-error
+except ImportError:  # pragma: no cover
+    win32security = None  # type: ignore
+
+
+def chmod(file_path, mode):
+    # type: (str, int) -> None
+    """
+    Apply a POSIX mode on given file_path:
+        * for Linux, the POSIX mode will be directly applied using chmod,
+        * for Windows, the POSIX mode will be translated into a Windows DACL that make sense for
+          Certbot context, and applied to the file using kernel calls.
+    :param str file_path: Path of the file
+    :param int mode: POSIX mode to apply
+    """
+    if not win32security:
+        os.chmod(file_path, mode)
+    else:
+        _apply_win_mode(file_path, mode)
+
+
+def _apply_win_mode(file_path, mode):
+    # Resolve symbolic links
+    if os.path.islink(file_path):
+        link_path = file_path
+        file_path = os.readlink(file_path)
+        if not os.path.isabs(file_path):
+            file_path = os.path.join(os.path.dirname(link_path), file_path)
+    # Get owner sid of the file
+    security = win32security.GetFileSecurity(file_path, win32security.OWNER_SECURITY_INFORMATION)
+    user = security.GetSecurityDescriptorOwner()
+
+    # New DACL, that will overwrite existing one (including inherited permissions)
+    dacl = _generate_dacl(user, mode)
+
+    # Apply the new DACL
+    security.SetSecurityDescriptorDacl(1, dacl, 0)
+    win32security.SetFileSecurity(file_path, win32security.DACL_SECURITY_INFORMATION, security)
+
+
+def _generate_dacl(user_sid, mode):
+    analysis = _analyze_mode(mode)
+
+    # Get standard accounts sid
+    system = win32security.ConvertStringSidToSid('S-1-5-18')
+    admins = win32security.ConvertStringSidToSid('S-1-5-32-544')
+    everyone = win32security.ConvertStringSidToSid('S-1-1-0')
```

Comment:
You want a joke ? Name accounts in Windows are translated in the current OS language. Means that `win32security.LookupAccountName ("", "Everyone")` does work only on a system configured to use english language. On my french machine, it would be `win32security.LookupAccountName ("", "Tout le monde")`.

SID are the only way to get consistently a particular account. On top of that, some SID are so-called "well-known". It means their value and what they represent is always the same. I put a link for that in the comments.